Objects/typeobject.c

```diff
@@ -9358,14 +9358,15 @@ super_repr(PyObject *self)
 }
 
 static PyObject *
-super_getattro(PyObject *self, PyObject *name)
+do_super_lookup(superobject *su, PyTypeObject *su_type, PyObject *su_obj,
```

Comment:
Could we filter out the `__class__` attribute earlier? Then the superobject arg can be dropped.
E.g. don't emit `LOAD_SUPER_ATTR` for the `__class__` attribute.
`super_getattro` would also need to be changed.

Reply:
Avoiding `LOAD_SUPER_ATTR` when the argument is `__class__` is possible, but it wouldn't allow dropping the `superobject` argument. It is also used in the `skip:` case below (the fallback to regular getattr on the `super` object), which in most cases has the effect of just generating an `AttributeError`, but in principle could also be used to fetch actual attributes of a `super` object (e.g. `__self__` and `__self_class__`).

Even if we can't get rid of the argument, it would still be nice to move this check out of the fast path, so I'll look at making that change.